lang/java/reef-io/src/main/java/org/apache/reef/io/network/NetworkServiceClient.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.io.network;
+
+import org.apache.reef.exception.evaluator.NetworkException;
+import org.apache.reef.io.network.impl.DefaultNetworkServiceClientImpl;
+import org.apache.reef.tang.annotations.DefaultImplementation;
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.wake.Identifier;
+import org.apache.reef.wake.remote.Codec;
+import org.apache.reef.wake.remote.transport.LinkListener;
+
+@DefaultImplementation(DefaultNetworkServiceClientImpl.class)
+/**
+ * NetworkServiceClient.
+ *
+ * NetworkServiceClient is a service which is designed for communicating messages with each other.
+ * Users can send messages with NetworkServiceClient, by registering their Codec, EventHandler, and LinkListener.
+ */
+public interface NetworkServiceClient extends AutoCloseable {
```

Comment:
I thought you used NetworkServiceClient since we cannot use the name
NetworkService. NetworkService is good.

On Fri, Jul 10, 2015 at 5:05 PM, Tae-Geon Um <notifications@github.com>
wrote:

> In
> lang/java/reef-io/src/main/java/org/apache/reef/io/network/NetworkServiceClient.java
> <https://github.com/apache/incubator-reef/pull/255#discussion_r34334653>:
>
> > +import org.apache.reef.exception.evaluator.NetworkException;
> > +import org.apache.reef.io.network.impl.DefaultNetworkServiceClientImpl;
> > +import org.apache.reef.tang.annotations.DefaultImplementation;
> > +import org.apache.reef.wake.EventHandler;
> > +import org.apache.reef.wake.Identifier;
> > +import org.apache.reef.wake.remote.Codec;
> > +import org.apache.reef.wake.remote.transport.LinkListener;
> > +
> > +@DefaultImplementation(DefaultNetworkServiceClientImpl.class)
> > +/**
> > + * NetworkServiceClient.
> > + *
> > + * NetworkServiceClient is a service which is designed for communicating messages with each other.
> > + * Users can send messages with NetworkServiceClient, by registering their Codec, EventHandler, and LinkListener.
> > + */
> > +public interface NetworkServiceClient extends AutoCloseable {
>
> Thanks for your suggestion. However, I'm worried that
> NetworkConnectionFactoryProvider seems to limit the behavior of this
> class. This class not only provides ConnectionFactory, but also register
> an identifier, in order to connect with each other. This class is
> responsible for handling encode/decode messages, routing messages to the
> corresponding ConnectionFactory.
>
> Do you think NetworkService is not good? how about NetworkManagerService,
> NetworkConnectionService or NetworkMessagingService?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/255/files#r34334653>.
>



-- 
Byung-Gon Chun